test/server.cpp

```diff
@@ -456,6 +455,16 @@ TEST_F(ServerTest, 400)
   }
 }
 
+TEST_F(ServerTest, EmptyPatternSearchDoesNotError)
+{
+  // Empty pattern
+  auto r = zfs1_->GET("/ROOT%23%3F/search?content=zimfile");
+  EXPECT_EQ(200, r->status);
+  // Whitespace-only pattern (should be trimmed)
+  r = zfs1_->GET("/ROOT%23%3F/search?content=zimfile&pattern=%20%20");
+  EXPECT_EQ(200, r->status);
+}
```

Comment:
Why is this change in this commit? What does it have to do with the modification in the frontend? Would this new unit-test pass with code state corresponding to this commit?